### PR TITLE
Reverts ...args

### PR DESCRIPTION
**Pull request by dumganhar**
Depends on https://github.com/cocos-creator/engine/pull/2033

Fixed https://github.com/cocos-creator/fireball/issues/6428

### CODE REVIEW

File: cocos/scripting/js-bindings/script/jsb_opengl.js [hunk 1]
```diff
@@ -353,11 +353,11 @@ gl.getAttachedShaders = function(program) {
 // void texImage2D(GLenum target, GLint level, GLenum internalformat, GLenum format, GLenum type, HTMLImageElement image); // May throw DOMException
 // void texImage2D(GLenum target, GLint level, GLenum internalformat, GLenum format, GLenum type, HTMLCanvasElement canvas); // May throw DOMException
 // void texImage2D(GLenum target, GLint level, GLenum internalformat, GLenum format, GLenum type, HTMLVideoElement video); // May throw DOMException
-gl.texImage2D = function (...args) {
-    if (args.length < 9) {
-        throw "texImage2D: Unsupported number of parameters:" + args.length;
-    }
-    gl._texImage2D.apply(gl, args);
+gl.texImage2D = function() {
+    if( arguments.length !=  9)
```
Comment (jareguo) [inline cocos/scripting/js-bindings/script/jsb_opengl.js]: length 判断方式确定也要 revert 吗？

File: cocos/scripting/js-bindings/script/jsb_prepare.js [hunk 2]
```diff
@@ -147,20 +147,10 @@ cc.Class.extend = function (prop) {
...
-                    case 0: this.ctor(); break;
-                    case 1: this.ctor(args[0]); break;
-                    case 2: this.ctor(args[0], args[1]); break;
-                    case 3: this.ctor(args[0], args[1], args[2]); break;
-                    case 4: this.ctor(args[0], args[1], args[2], args[3]); break;
-                    case 5: this.ctor(args[0], args[1], args[2], args[3], args[4]); break;
-                    default: this.ctor.apply(this, args);
-                }
-            }
+            this.ctor && this.ctor.apply(this, arguments);
```
Comment (jareguo) [inline cocos/scripting/js-bindings/script/jsb_prepare.js]: 这里保留一下 switch 吧

Comment (dumganhar) [reply cocos/scripting/js-bindings/script/jsb_prepare.js]: 是性能考虑么？

File: cocos/scripting/js-bindings/script/jsb_opengl.js (hunk 1 above)
Comment (dumganhar) [reply cocos/scripting/js-bindings/script/jsb_opengl.js]: 原来不是也判断 if (args.length < 9) {么，没改语义啊。

Comment (jareguo) [reply cocos/scripting/js-bindings/script/jsb_opengl.js]: != 9 也有可能 > 9 吧

File: cocos/scripting/js-bindings/script/jsb_prepare.js (hunk 2 above)
Comment (jareguo) [reply cocos/scripting/js-bindings/script/jsb_prepare.js]: 是的，call / apply 能避免就避免

**Comment by dumganhar** [comment]
@jareguo , all issues you replied were fixed. Please review again.

**Comment by jareguo** [comment]
LGTM

**Pull request merged**